### PR TITLE
fix: cleanup languageId from NlpSamples on language deletion

### LINKED ISSUE #781: 🤔 [ISSUE] - delete language need to update related collections
### Description
We can notice that the language in included in more that one collection.
Currently, when you delete a language the other linked collection will preserve the invalid ObjectId.
The goal is to update the related collections to remove the language id from him (NlpSample)

### Reproduction steps

```bash
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
```

### Logs

```bash

```

### Browsers

_No response_

### OS

_No response_

### PR DESCRIPTION
**Pull request by abdou6666**
# Motivation

On language deletion we must cleanup the `languageId` in `NlpSamples`

Fixes #781

# Type of change:

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] I have performed a self-review of my own code

## Summary by CodeRabbit

- **Refactor**
  - Enhanced the language removal process to better support bulk deletions.
  - Improved data consistency by ensuring associated records are updated accurately when languages are removed.
  - Updated event handling for language deletions to operate at a different timing.
  - Removed obsolete pre-deletion logic for languages.

### CODE REVIEW

**Review by MohamedAliBouhaouala — APPROVED** [review]

**Review by IkbelTalebHssan — APPROVED** [review]

File: api/src/i18n/repositories/language.repository.ts [hunk 1]
```diff
@@ -43,14 +43,17 @@ export class LanguageRepository extends BaseRepository<
       Document<Language, any, any>,
       unknown,
       Language,
-      'deleteOne' | 'deleteMany'
+      'deleteOne'
```
Comment (marrouchi) [inline api/src/i18n/repositories/language.repository.ts:46]: Let's remove this function and use OnEvent('hook:language:preDelete') in the NlpSampleService.
const languages = this.languageService.find(criteria)
await this.nlpSampleService.updateMany(criteria, { $set:{ language: null } })

**Review by marrouchi — CHANGES_REQUESTED** [review]

**Comment by marrouchi** [comment]
@coderabbitai review

File: api/src/nlp/services/nlp-sample.service.ts [hunk 2]
```diff
@@ -235,11 +237,32 @@ export class NlpSampleService extends BaseService<
...
+      },
+    );
+    const deletedLanguagesIds = deletedLanguages.map(
+      (deletedLanguage) => deletedLanguage.id,
+    );
     await this.updateMany(
       {
-        language: language.id,
+        language: {
+          $in: deletedLanguagesIds,
```
Comment (marrouchi) [inline api/src/nlp/services/nlp-sample.service.ts:264]: What will be the behavior if deletedLanguagesIds is an empty array ? Ensuring the unit test covers this edge case can be beneficial.

**Review by marrouchi — CHANGES_REQUESTED** [review]
@abdou6666 Can you please rebase the PR ?

File: api/src/nlp/services/nlp-sample.service.ts (hunk 2 above)
Comment (yassinedorbozgithub) [reply api/src/nlp/services/nlp-sample.service.ts:262]: I think that we need to add an if condition to check if deletedLanguagesIds that has elements execute the the update

**Review by yassinedorbozgithub — APPROVED** [review]
